contracts/OlasMech.sol

```diff
@@ -90,118 +97,88 @@ abstract contract OlasMech is Mech, IErrorsMech, ImmutableStorage {
 
     /// @dev Registers a request.
-    /// @param requester Requester address.
-    /// @param requestId Request Id.
-    /// @param data Self-descriptive opaque data-blob.
+    /// @param requestIds Set of request Ids.
+    /// @param datas Set of corresponding self-descriptive opaque data-blobs.
     function _request(
-        address requester,
-        uint256 requestId,
-        bytes memory data
+        uint256[] memory requestIds,
+        bytes[] memory datas
     ) internal virtual {
-        mapUndeliveredRequestsCounts[requester]++;
-        // Record the requestId => sender correspondence
-        mapRequestAddresses[requestId] = requester;
-
-        // Record the request Id in the map
-        // Get previous and next request Ids of the first element
-        uint256[2] storage requestIds = mapRequestIds[0];
-        // Create the new element
-        uint256[2] storage newRequestIds = mapRequestIds[requestId];
-
-        // Previous element will be zero, next element will be the current next element
-        uint256 curNextRequestId = requestIds[1];
-        newRequestIds[1] = curNextRequestId;
-        // Next element of the zero element will be the newly created element
-        requestIds[1] = requestId;
-        // Previous element of the current next element will be the newly created element
-        mapRequestIds[curNextRequestId][0] = requestId;
+        uint256 numRequests = requestIds.length;
+
+        for (uint256 i = 0; i < requestIds.length; ++i) {
+            uint256 requestId = requestIds[i];
+
+            // TODO optimize or remove?
+            // Record the request Id in the map
+            // Get previous and next request Ids of the first element
+            uint256[2] storage requestIdLinks = mapRequestIds[0];
+            // Create the new element
+            uint256[2] storage newRequestIdLinks = mapRequestIds[requestId];
+
+            // Previous element will be zero, next element will be the current next element
+            uint256 curNextRequestIdLink = requestIdLinks[1];
+            newRequestIdLinks[1] = curNextRequestIdLink;
+            // Next element of the zero element will be the newly created element
+            requestIdLinks[1] = requestId;
+            // Previous element of the current next element will be the newly created element
+            mapRequestIds[curNextRequestIdLink][0] = requestId;
+
+            emit Request(requestId, datas[i]);
+        }
 
         // Increase the number of undelivered requests
-        numUndeliveredRequests++;
+        numUndeliveredRequests += numRequests;
         // Increase the total number of requests
-        numTotalRequests++;
-
-        emit Request(requester, requestId, data);
-    }
-
-    /// @dev Cleans the request info from all the relevant storage.
-    /// @param requester Requester address.
-    /// @param requestId Request Id.
-    function _cleanRequestInfo(address requester, uint256 requestId) internal virtual {
```

Comment:
Became obsolete in this form, as there are no push from MM anymore. This logic went into pre-deliver of requests to count the number of self and not self ones.